crates/turborepo-lib/src/turbo_json/parser.rs

```diff
@@ -0,0 +1,556 @@
+use std::{collections::BTreeMap, sync::Arc};
+
+use biome_console::{markup, ColorMode, ConsoleExt, EnvConsole};
+use biome_deserialize::{
+    json::deserialize_from_json_str, Deserializable, DeserializableValue,
+    DeserializationDiagnostic, DeserializationVisitor, Text, VisitableType,
+};
+use biome_diagnostics::{DiagnosticExt, PrintDiagnostic};
+use biome_json_parser::JsonParserOptions;
+use biome_json_syntax::TextRange;
+use miette::Diagnostic;
+use thiserror::Error;
+
+use crate::{
+    cli::OutputLogsMode,
+    config::ConfigurationOptions,
+    run::task_id::TaskName,
+    turbo_json::{Pipeline, RawTaskDefinition, RawTurboJson, SpacesJson, Spanned},
+};
+
+#[derive(Debug, Error, Diagnostic)]
+pub enum Error {
+    #[error("failed to parse turbo.json")]
+    Parse {
+        diagnostics: Vec<biome_diagnostics::Error>,
+    },
+}
+
+fn print_diagnostics(diagnostics: &[biome_diagnostics::Error], color: bool) {
+    let color_mode = if color {
+        ColorMode::Enabled
+    } else {
+        ColorMode::Disabled
+    };
+    let mut console = EnvConsole::new(color_mode);
```

Comment:
Can we use biome without a new console/color mode abstraction? Is there a path to either swap all of our current usage to this or vice versa?

I'd rather not have multiple console/color abstractions in the codebase.

Reply:
nvm, tackled